src/reducers/reducers.js

```diff
@@ -0,0 +1,60 @@
+import { combineReducers } from 'redux';
+import { ActionTypes } from '../actions';
+
+const initialState = {
+  articles: [],
+  page: 0,
+  isLoading: false,
+  valid: false,
+};
+
+function articles(state = initialState.articles, action) {
+  switch (action.type) {
+    case ActionTypes.FETCH_NEWS_SUCCESS:
+      if (action.payload.page === 0) {
+        return action.payload.articles;
+      }
+      return [...state, ...action.payload.articles];
+    default:
+      return state;
+  }
+}
+
+function page(state = initialState.page, action) {
+  switch (action.type) {
+    case ActionTypes.FETCH_NEWS_SUCCESS:
+      return action.payload.page;
+    default:
+      return state;
+  }
+}
+
+function isLoading(state = initialState.isLoading, action) {
+  switch (action.type) {
+    case ActionTypes.FETCH_NEWS_BEGIN:
+      return true;
+    case ActionTypes.FETCH_NEWS_SUCCESS:
+    case ActionTypes.FETCH_NEWS_ERROR:
+      return false;
+    default:
+      return state;
+  }
+}
+
+function valid(state = initialState.valid, action) {
+  switch (action.type) {
+    case ActionTypes.INVALIDATE_NEWS:
+      return false;
+    case ActionTypes.FETCH_NEWS_SUCCESS:
+      return true;
+    default:
+      return state;
+  }
+}
```

Comment:
Merge above functions into 1 reducer, this is correct but a bit unconventional (all properties are describing the same resource so it is easier to maintain it if one reducer is handling it). You can easily manage state for `ActionTypes.FETCH_NEWS_SUCCESS` for all properties:

```
case ActionTypes.FETCH_NEWS_SUCCESS: {
  const { page, articles } = action.payload;
  const newArticles = page === 0 ? articles : [...state.articles, articles];
  
  return {
    ...state,
    valid: true,
    page,
    isLoading: false,
    articles: newArticles,
  }
}
```